### PR TITLE
Symmetric crypto tests

### PR DESCRIPTION
**Pull request by JustinMoss**

### CODE REVIEW

File: Rock.Encryption/Symmetric/CryptoConfiguration.cs [hunk 1]
```diff
@@ -12,6 +12,6 @@ public class CryptoConfiguration
         /// Gets the collection of credentials that will be available for encryption or
         /// decryption operations.
         /// </summary>
-        public List<Credential> Credentials { get; set; } = new List<Credential>();
+        public List<ICredential> Credentials { get; set; } = new List<ICredential>();
```
Comment (bfriesen) [inline Rock.Encryption/Symmetric/CryptoConfiguration.cs]: I don't think this class will serialize if it depends on the interface. @JustinMoss: can you confirm?

File: RockLib.Encryption.Tests/SymmetricCryptoTests.cs [hunk 2]
```diff
@@ -69,17 +85,27 @@ public void CanCreateSymmetricCryptoByEncryptionSettings()
...
+            });
 
             var crypto = new SymmetricCrypto();
             crypto.EncryptionSettings = new CryptoConfiguration
             {
-                Credentials = new List<ICredential>
+                Credentials = new List<Credential> 
                 {
-                    credentialMock.Object
+                    new Credential
```
Comment (bfriesen) [inline RockLib.Encryption.Tests/SymmetricCryptoTests.cs]: The part here with LateBoundConfigurationSection is so awkward. I wonder if we should add a constructor to the Credential class that hides that complexity. I think your implementation here would be the right implementation for the constructor too.

``` c#
public Credential(SymmetricAlgorithm algorithm, int ivSize, byte[] key)
{
    // TODO: argument checking

    Algorithm = algorithm;
    IVSize = ivSize;

    var builder = new ConfigurationBuilder();
    builder.AddInMemoryCollection(new Dictionary<string, string> { { "Section:Value", Convert.ToBase64String(key) } });

    Key = new LateBoundConfigurationSection<IProtectedValue>
    {
        Type = "RockLib.DataProtection.UnprotectedBase64Value, RockLib.DataProtection",
        Value = builder.Build().GetSection("Section")
    };
}
```

Thechange would mean this line would simply be this:

``` c#
new Credential(SymmetricAlgorithm.Aes, 16, Convert.FromBase64String("1J9Og/OaZKWdfdwM6jWMpvlr3q3o7r20xxFDN7TEj6s=")
```

**Comment by derikwhittaker** [comment]
have we addressed Brian's questions?  If so than I am good w/ this and will merge when i get confirmation.

File: Rock.Encryption/Symmetric/Credential.cs [hunk 3]
```diff
@@ -35,6 +36,32 @@ public Credential()
...
+            if (ivSize <= 0) throw new ArgumentOutOfRangeException(nameof(ivSize), $"{nameof(ivSize)} must be greater than 0");
+            if (key == null) throw new ArgumentNullException(nameof(key));
+
+            Algorithm = algorithm;
+            IVSize = ivSize;
+
+            var builder = new ConfigurationBuilder();
+            builder.AddInMemoryCollection(new Dictionary<string, string> { { "Section:Value", Convert.ToBase64String(key) } });
+
+            Key = new LateBoundConfigurationSection<IProtectedValue>
```
Comment (derikwhittaker) [inline Rock.Encryption/Symmetric/Credential.cs]: seems very confusing to me that we have Key and _key but they are both different types.

I just assumed that _key was the backing field for Key, but this not correct.

Comment (JustinMoss) [reply Rock.Encryption/Symmetric/Credential.cs]: In a way, it is the backing field.  In Key, it's still a config section, in _key, it's extracted from the config section for easier use later.  If we want to discuss a better way to do this, I'm fine with it, but this is how it was before any of my changes, so it should probably be a seperate PR.

Comment (derikwhittaker) [reply Rock.Encryption/Symmetric/Credential.cs]: Sounds good, thank you for addressing this.  I say leave it for now

**Pull request merged**